pinot-core/src/main/java/org/apache/pinot/core/util/SegmentProcessorAvroUtils.java

```diff
@@ -65,6 +66,10 @@ public static GenericData.Record convertGenericRowToAvroRecord(GenericRow generi
         if (value instanceof byte[]) {
           value = ByteBuffer.wrap((byte[]) value);
         }
+        if (value instanceof BigDecimal) {
+          // Convert BigDecimal to String for Avro compatibility
```

Comment:
Please ignore this class. Here I'm hotfixing big decimals so I can ingest them during tests, but the proper fix is in https://github.com/apache/pinot/pull/16210, which should be merged first